### PR TITLE
Adjust F# code formatting doc

### PR DESCRIPTION
**Pull request by cartermp**
Fixes a few things from #5744 and moves some naming and whitespace guidelines up near the top.

### CODE REVIEW

File: docs/fsharp/style-guide/formatting.md [hunk 1]
```diff
@@ -25,6 +25,177 @@ When indentation is required, you must use spaces, not tabs. At least one space
...
+
+    ...
+```
+
+### Use PascalCase for type declarations, members, and labels
+
+Classes, interfaces, structs, enumerations, delegates, records, and discriminated unions should all be named with PascalCase. Members within types and labels for records and discriminated unions should also use PascalCase.
+
+```fsharp
+type IMyInterface =
```
Comment (BillWagner) [inline docs/fsharp/style-guide/formatting.md]: Do you want to recommend that interfaces start with `I`?

File: docs/fsharp/style-guide/formatting.md [hunk 2]
```diff
@@ -25,6 +25,177 @@ When indentation is required, you must use spaces, not tabs. At least one space
...
+Some exceptions includes interoperating with native components, where underscores are very common.
+
+### Use standard F# operators
+
+The following operators are defined in the F# standard library and should be used instead of defining equivalents. Using these operators is recommended as it tends to make code more readable and idiomatic. Developers with a background in OCaml or other functional programming language may be accustomed to different idioms. The following list summarizes the recommended F# operators.
+
+```fsharp
+x |> f // Forward pipeline
+f >> g // Forward composition
+x |> ignore // Throwing away a value
```
Comment (BillWagner) [inline docs/fsharp/style-guide/formatting.md]: nit: consider "discard a value" to avoid similarity to `throw`

**Review by BillWagner — APPROVED** [review]
This LGTM.  I had a couple suggestions, if you want to adopt them.

:shipit: when you are ready.

File: docs/fsharp/style-guide/formatting.md (hunk 1 above)
Comment (cartermp) [reply docs/fsharp/style-guide/formatting.md]: Yep. We follow .NET naming guidelines here.

**Pull request merged**